framework/src/utils/MathUtils.C

```diff
@@ -14,181 +14,152 @@ namespace MathUtils
-              (x - tol) * (x - tol) * (x - tol) / (3.0 * tol * tol * tol) -
-              (x - tol) * (x - tol) * (x - tol) * (x - tol) / (4.0 * tol * tol * tol * tol) +
-              (x - tol) * (x - tol) * (x - tol) * (x - tol) * (x - tol) /
-                  (5.0 * tol * tol * tol * tol * tol) -
-              (x - tol) * (x - tol) * (x - tol) * (x - tol) * (x - tol) * (x - tol) /
-                  (6.0 * tol * tol * tol * tol * tol * tol);
-    else
-      value = std::log(x);
-  }
-  else if (order == 1)
-  {
-    if (x < tol)
-      value = 1.0 / tol - 2.0 * (x - tol) / (2.0 * tol * tol) +
-              3.0 * (x - tol) * (x - tol) / (3.0 * tol * tol * tol) -
-              4.0 * (x - tol) * (x - tol) * (x - tol) / (4.0 * tol * tol * tol * tol) +
-              5.0 * (x - tol) * (x - tol) * (x - tol) * (x - tol) /
-                  (5.0 * tol * tol * tol * tol * tol) -
-              6.0 * (x - tol) * (x - tol) * (x - tol) * (x - tol) * (x - tol) /
-                  (6.0 * tol * tol * tol * tol * tol * tol);
-    else
-      value = 1.0 / x;
-  }
-  else if (order == 2)
-  {
-    if (x < tol)
-      value = -2.0 * 1.0 / (2.0 * tol * tol) + 3.0 * 2.0 * (x - tol) / (3.0 * tol * tol * tol) -
-              4.0 * 3.0 * (x - tol) * (x - tol) / (4.0 * tol * tol * tol * tol) +
-              5.0 * 4.0 * (x - tol) * (x - tol) * (x - tol) / (5.0 * tol * tol * tol * tol * tol) -
-              6.0 * 5.0 * (x - tol) * (x - tol) * (x - tol) * (x - tol) /
-                  (6.0 * tol * tol * tol * tol * tol * tol);
-    else
-      value = -1.0 / (x * x);
-  }
-  else if (order == 3)
+  const auto c1 = [&]() { return 1.0 / (3.0 * tol * tol * tol); };
+  const auto c2 = [&]() { return -3.0 / (2.0 * tol * tol); };
+  const auto c3 = [&]() { return 3.0 / tol; };
+  const auto c4 = [&]() { return std::log(tol) - 11.0 / 6.0; };
+
+  switch (deriv)
   {
-    if (x < tol)
-      value = 3.0 * 2.0 * 1.0 / (3.0 * tol * tol * tol) -
-              4.0 * 3.0 * 2.0 * (x - tol) / (4.0 * tol * tol * tol * tol) +
-              5.0 * 4.0 * 3.0 * (x - tol) * (x - tol) / (5.0 * tol * tol * tol * tol * tol) -
-              6.0 * 5.0 * 4.0 * (x - tol) * (x - tol) * (x - tol) /
-                  (6.0 * tol * tol * tol * tol * tol * tol);
-    else
-      value = 2.0 / (x * x * x);
+    case 0:
+      return c1() * x * x * x + c2() * x * x + c3() * x + c4();
+
+    case 1:
+      return 3.0 * c1() * x * x + 2.0 * c2() * x + c3();
+
+    case 2:
+      return 6.0 * c1() * x + 2.0 * c2();
+
+    case 3:
+      return 6.0 * c1();
   }
-  return value;
+
+  mooseError("Unsupported derivative order ", deriv);
+}
+
+Real
+poly4Log(Real x, Real tol, int deriv)
+{
+  if (x >= tol)
+    return plainLog(x, deriv);
+
+  switch (deriv)
+  {
+    case 0:
+      return std::log(tol) + (x - tol) / tol - (x - tol) * (x - tol) / (2.0 * tol * tol) +
+             (x - tol) * (x - tol) * (x - tol) / (3.0 * tol * tol * tol) -
```

Comment:
`Utility::pow<3>(x-tol)`?

Reply:
I'll put that in for clarity, but even with that magic template you need 2 multiplications to get to power 3. It only starts helping >= power 4

Reply:
You could even do `3.0*Utility::pow<3>((x-tol)/tol))`

Reply:
Or even super complex by nesting all of the operators...(x-tol)/tol * (1 + (x-tol)/tol * (-2 + 3 * (x-tol)/tol)), or something like that

Reply:
Yeah, I think I'll leave it as is for now. I think getting too cute doesn't help readability.